### PR TITLE
zephyr: add a generic Zephyr platform

### PR DESCRIPTION
**Pull request by lyakh**
When built with Zephyr, SOF ideally should be a fully platform-agnostic application. All platform-specific code should reside in Zephyr itself, including core architecture and platform infrastructures and drivers. To prepare for this transition we add a generic SOF-Zephyr platform. It is based on the current Tiger Lake architecture and should be usable for cAVS 2.5 and newer systems.

This commit has been tested as a replacement for the usual Tiger Lake configuration with Zephyr.

### CODE REVIEW

File: src/platform/zephyr/include/platform/drivers/alh.h [hunk 1]
```diff
@@ -0,0 +1,95 @@
+/* SPDX-License-Identifier: BSD-3-Clause
+ *
```
Comment (dbaluta) [inline src/platform/zephyr/include/platform/drivers/alh.h:2]: alh is something intel specific right? In the future under SOF_ZEPHYR should be some sort of platform independent abstractions right?

**Review by dbaluta — APPROVED** [review]

**Review by lgirdwood — APPROVED** [review]

**Comment by marc-hb** [comment]
The formal approvals feel a bit premature to me considering no Zephyr build succeeded...

**Comment by lyakh** [comment]
The CI zephyr-build failure is unrelated:
```
--- hal_xtensa: fetching, need revision 0e577021bb66e644afd067cd9f7c71ab11b62b3d
error: RPC failed; curl 56 GnuTLS recv error (-54): Error in the pull function.
fatal: the remote end hung up unexpectedly
fatal: early EOF
fatal: index-pack failed
ERROR: update failed for project hal_xtensa
```

**Review by lgirdwood — APPROVED** [review]

**Comment by lgirdwood** [comment]
Good start, this should start reducing in size as IP code is upstreamed into Zephyr so that we are left with minimal.

**Pull request merged**